lmp/model/_transformer.py

```diff
@@ -0,0 +1,460 @@
+r"""Transformer Language Model with Transformer's encoder architecture."""
+
+
+import math
+import argparse
+from typing import ClassVar, Dict, Optional
+
+import torch
+import torch.nn as nn
+
+from lmp.model._rnn import RNNModel
+from lmp.tknzr._base import BaseTknzr
+
+
+class PositionalEncoding(nn.Module):
+    r"""Positional Encoding.
+
+    In order for the model to make use of the order of the sequence,
+    add ``positional encodings`` to the input embeddings.
+
+    .. math::
+
+            \begin{align}
```

Comment:
Only need one indentation for each `.rst` block.

Reply:
You use `d_model` in the formulation but use `d_hid` throughout your code instead.
Be consistent.